docs/plugins/plugin-writer/developer-setup.rst

```diff
@@ -0,0 +1,110 @@
+Developer Setup
+===============
+
+Each release of ``pulpcore`` on PyPI is released as a Vagrant box on `vagrantcloud.com <https://app.vagrantup.com/pulp/boxes/pulpcore>`_.
+To get started, you need to install ``Vagrant``. You can refer `this <https://docs.pulpproject.org/dev-guide/contributing/dev_setup.html#vagrant>`_.
+For backends, you can opt for either `libvirt <https://docs.pulpproject.org/dev-guide/contributing/dev_setup.html#prerequisites-for-libvirt>`_ or 
+`Virtualbox <https://www.virtualbox.org/wiki/Downloads>`_.
+
+Create a ``Vagrantfile`` with below configuration in an empty project directory::
```

Comment:
please, put the "code" block into a file ``pulp/docs/plugins/plugin-writer/Vagrantfile`` and link it here to be downloadable:

```rest
One can use the :download:`Vagrantfile <Vagrantfile>` as a starting point and modify it as necessary.

```